### PR TITLE
Real fix for the pod name regex

### PR DESCRIPTION
**Pull request by krzyzacy**
#2552 

ref https://github.com/kubernetes/kubernetes/blob/616038db1b0d1e852b4a3d10c8c512a052f91fba/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L94

### CODE REVIEW

**Comment by krzyzacy** [comment]
it should expect to fail. #2552 fixes the test.

/assign @spxtr

File: prow/config/jobs_test.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 	"testing"
 )
 
-var podRe = regexp.MustCompile(`[a-z0-9]([-a-z0-9]*[a-z0-9])?`)
+var podRe = regexp.MustCompile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`)
```
Comment (spxtr) [inline prow/config/jobs_test.go]: ahhhhh welp

**Comment by krzyzacy** [comment]
@k8s-bot bazel test this

**Pull request merged**